File: php/tests/SendTest.php
<?php

class SendTest extends PHPUnit_Framework_TestCase
{
    private $_xkey;
    private $_driver;

    private $_message;
    private $_messageId;
    private $_targetId;

    /*
    * driver creation is tested in another file
    */
    public function __construct()
    {
        $this->_xkey = 'xkey';
        $this->_driver = new Driver(['xKey' => $this->_xkey, 'url' => 'http://127.0.0.1:8888/']);
        $this->_message = [
            'content'       => [
                'html'      => 'PLEASE REPORT TO SOMEONE IF YOU RECEIVED THIS EMAIL',
                'text'      => 'PLEASE REPORT TO SOMEONE IF YOU RECEIVED THIS EMAIL'
            ],
            'header'        => [
                'subject'   => 'subject of the message',
                'mailFrom'  => 'mail@address.com',
                'replyTo'   => 'mail@return.com'
            ]
        ];
        $this->_messageId = '000QF0';
        $this->_targetId = '000OG3XW';
    }
    /**
     * It should send a message to a target
     *
     */
    public function testMessageSending()
    {
        $response = $this->_driver->send->message($this->_messageId, $this->_targetId);
        $this->assertEquals($response['statusCode'], 204);
    }
    /**
     * It should send a message with a different
     * content then the one in it to a target
     *
     */
    public function testMessageSendingWithDifferentContent()
    {
        $response = $this->_driver->send->message($this->_messageId, $this->_targetId, $this->_message);
        $this->assertEquals($response['statusCode'], 204);
    }
}

?>

File: java/src/test/java/route/SendTest.java
package route;

import org.json.JSONObject;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class SendTest {

    private Driver driver;
    private JSONObject message = new JSONObject("{" +
            "'content': {\n" +
            "   'html': 'PLEASE REPORT TO SOMEONE IF YOU RECEIVED THIS EMAIL',\n" +
            "   'text': 'PLEASE REPORT TO SOMEONE IF YOU RECEIVED THIS EMAIL'\n" +
            "},\n" +
            "'header': {\n" +
            "   'subject': 'subject of the message',\n" +
            "   'mailFrom': 'mail@address.com',\n" +
            "   'replyTo': 'mail@return.com'\n" +
            "}\n" +
            "}");
    private String messageId = "000QF0";
    private String targetId = "000OG3XW";

    @Before
    public void setUp() {
        JSONObject config = new JSONObject();

        config.put("xkey", "my_xkey");
        config.put("url", "http://127.0.0.1:8888/");
        driver = new Driver(config);
    }

    @Test
    public void testSend() {
        JSONObject response = this.driver.send.message(messageId, targetId);
        assertEquals(response.getInt("statusCode"), 204);
    }

    @Test
    public void testSendWithNewContent() {
        JSONObject response = this.driver.send.message(messageId, targetId, message);
        assertEquals(response.getInt("statusCode"), 204);
    }

}
